frontend/src/features/settings/components/creditAccount/UserTable.tsx

```diff
@@ -0,0 +1,197 @@
+
+  const [isRemoveModalOpen, setIsRemoveModalOpen] = useState(true);
+  const [userIds, setUserIds] = useState<number[]>([]);
+  const [rowSelection, setRowSelection] = useState<RowSelectionState>({});
+  const hasNoRowsSelected = Object.keys(rowSelection).length === 0;
+
+  /**
+   * Callback function for clicking on the 'remove user' button above the Table.
+   */
+  const handleRemoveButton = useCallback(() => {
+    setIsRemoveModalOpen(() => true);
+  }, []);
+
+  /**
+   * Retrieves user identifiers from the row selection state.
+   */
+  const getSelectedUsers = useCallback(() => {
+    return Object.keys(rowSelection).map((value: string) => Number(value));
+  }, [rowSelection]);
+
+  /**
+   * Close RemoveUserModal and clear row selection state
+   */
+  const handleConfirmRemove = async () => {
+    setIsRemoveModalOpen(() => false);
+    setRowSelection(() => {
+      return {};
+    });
+    refetch();
+  };
+
+  /**
+   * Function that clears the delete related states when the user clicks on cancel.
+   */
+  const handleCancelRemove = useCallback(() => {
+    setIsRemoveModalOpen(() => false);
+    setRowSelection(() => {
+      return {};
+    });
+  }, []);
+
+  useEffect(() => {
+    setUserIds(getSelectedUsers());
+  }, [rowSelection]);
+
+  const table = useMaterialReactTable({
+    ...defaultTableOptions,
+    columns: CreditAccountUserColumnsDefinition,
+    data: isAccountHolder ? creditAccountUsers : [accountHolder],
+    initialState: {
+      ...defaultTableInitialStateOptions,
+    },
+    state: {
+      ...defaultTableStateOptions,
+      showAlertBanner: isError,
+      showProgressBars: isLoading,
+      isLoading: isLoading,
+      rowSelection: rowSelection,
+    },
+    layoutMode: "grid",
+    enableGlobalFilter: false,
+    renderEmptyRowsFallback: () => <NoRecordsFound />,
+    enableRowSelection: showCheckboxes
+      ? (row: MRT_Row<CreditAccountUser>): boolean => {
+          if (row?.original?.userType === "HOLDER") {
+            return false;
+          }
+          return true;
+        }
+      : false,
+    onRowSelectionChange: setRowSelection,
+    getRowId: (originalRow: CreditAccountUser) => {
+      return originalRow.companyId;
+    },
+    displayColumnDefOptions: {
+      "mrt-row-actions": {
+        header: "",
+      },
+      "mrt-row-select": {
+        size: 1,
+      },
+    },
+    renderToolbarInternalActions: useCallback(
+      () => (
+        <Box className="toolbar__inner">
+          <Typography variant="h3" className="user-table__heading">
+            Credit Account Users
+          </Typography>
+          {canUpdateCreditAccount(userRoles) && (
+            <Button
+              className="user-table__button user-table__button--remove"
+              onClick={handleRemoveButton}
+              disabled={hasNoRowsSelected}
+            >
+              Remove User
+            </Button>
+          )}
+        </Box>
+      ),
+      [hasNoRowsSelected],
```

Comment:
I think you need to add handleRemoveButton and userRoles as part of the dependency array as well, and perhaps even better, maybe extract the canUpdateCreditAccount(userRoles) part outside of this callback function, and use its result here instead (and including it in the dependency array)